sql-plugin/src/main/java/com/nvidia/spark/rapids/JCudfUtil.java

```diff
@@ -62,7 +62,8 @@ public final class JCudfUtil {
   /**
    * The maximum buffer size allocated to copy JCudf row.
    */
-  public static final long JCUDF_MAX_DATA_BUFFER_LENGTH = Integer.MAX_VALUE;
+  public static final long JCUDF_MAX_DATA_BUFFER_LENGTH =
+      Integer.MAX_VALUE - (JCUDF_ROW_ALIGNMENT - 1);
```

Comment:
Why are we using `Integet.MAX_VALUE` as a limit? It is a limit on a cudf column due to `cudf::size_type` being a signed int, but is there a reason to limit the row here to an int? I'm not against this, just ensuring it isn't being coupled unnecessarily to `cudf::size_type`.

Reply:
During transitions, the rows are 8-bytes aligned. So, `Integer.MAX_VALUE - 7` is 8 bytes aligned.
Probably I should rename this to a `JCUDF_MAX_ROW_SIZE_LENGTH` because the bufferSize which is multiple rows can be larger.